mypy/subtypes.py

```diff
@@ -263,8 +257,7 @@ def visit_overloaded(self, left: Overloaded) -> bool:
         elif isinstance(right, TypeType):
             # All the items must have the same type object status, so
             # it's sufficient to query only (any) one of them.
-            # This is unsound, we don't check the __init__ signature.
```

Comment:
I think this comment should stay here, since the ``__init__`` signature should be checked for all items in an overload.